### ISSUE TITLE
Could you please clarify the logic of eval_batch_size setting in FullSortEvalDataLoader and NegSampleEvalDataLoader

### ISSUE THREAD

**Issue by kobejie** [RUCAIBox/RecBole#1810]
Hi, I have two questions of setting eval_batch_size:

i) In FullSortEvalDataLoader, why the eval_batch_size is set differently for sequential models and other models. Per my understanding, it seems that the evaluation for sequential models is  _interaction-based_ while other models are _users-based_. What is the rationale behind such a design?

https://github.com/RUCAIBox/RecBole/blob/a757b7c31bfa407c82847b4a06c8580fa1a6b45e/recbole/data/dataloader/general_dataloader.py#L244-L253

ii) In NegSampleEvalDataLoader, what exactly is eval_batch_size set to? How can we intuitively set this parameter under 'uni100' or 'pop100'?

https://github.com/RUCAIBox/RecBole/blob/a757b7c31bfa407c82847b4a06c8580fa1a6b45e/recbole/data/dataloader/general_dataloader.py#L120-L138

**Comment by Sherry-XLL** [comment]
Hello @diesel248! Thanks for your attention to RecBole!

Whether the evaluation mode of the model is _interaction-based_ or _users-based_ can be determined by the user:

> group_by (str)decides how we group the data in .inter. Now we support two kinds of grouping strategies: ['user', 'none']. If the value of group_by is user, the data will be grouped by the column of USER_ID_FIELD and split in user dimension. If the value is none, the data won’t be grouped. The default value is user.

---

You can refer to [batch size adaptation](https://trello.com/c/e0R5hCn0/17-traineval-batch-size-adaptation) for more details about the batch size of different modes.

> Due to the memory limitation, we cannot load all the data at one time, so we need to conduct training and evaluation in the form of mini-batches.
> 
> Specifically, the number of batches = the total amount of data / batch_size. However, for the general and knowledge-aware models, the total amount of data for training and evaluation is different.
> 
> In the training phase, the total amount of data = the interacted user-item pairs + the negative pairs. The positive items = the split ratio * total data, while the negative ones come from the negative sampling module.
> 
> In the validation/test phase, the total amount of data depends on the evaluation mode. The total amount of data may be very large if the evaluation mode is full ranking rather than sampled-based ranking, because we need to calculate scores of each user with all items.
> 
> Therefore, in the full model of evaluation, the eval_batch_size needs to be adapted according to the number of items.
> 
> Related issues: [#544](https://github.com/RUCAIBox/RecBole/issues/544), [#678](https://github.com/RUCAIBox/RecBole/issues/678), [#958](https://github.com/RUCAIBox/RecBole/issues/958), [#1080](https://github.com/RUCAIBox/RecBole/discussions/1080), [#1093](https://github.com/RUCAIBox/RecBole/issues/1093), [#1556](https://github.com/RUCAIBox/RecBole/issues/1556), [#1615](https://github.com/RUCAIBox/RecBole/issues/1615)

---

You can also refer to our [FAQ page](https://trello.com/c/4VO0Z4uT/37-sampling-full-based-evaluation) for more details about evaluation in RecBole:

> In RecBole, we support both full and sampling-based evaluation. Generally speaking, when the number of items in the data set is very large, it will obviously waste a lot of time if we evaluate all items for each user. At this time, we will sample some negative examples for each positive example to speed up the evaluation process.
> 
> However, there is deviation between sampling-based metrics and full ranking evaluation as discussed in [On Sampling Top-K Recommendation Evaluation](https://dl.acm.org/doi/10.1145/3394486.3403262). When designing the architecture, we have taken into account that many recommendation papers are evaluated by sampling-based metrics. In order to make users adapt to the experimental environment of the original papers as much as possible when reproducing the results, we design two kinds of evaluation methods, sampling and full sorting, in the evaluation part. ([#1256](https://github.com/RUCAIBox/RecBole/issues/1256))
> 
> mode (str): decides the data range which we evaluate the model on. Now we support four kinds of evaluation mode: ['full','unixxx','popxxx','labeled']. full , unixxx and popxxx are designed for the evaluation on implicit feedback (data without label).
> 
> For implicit feedback, we regard the items with observed interactions as positive items and those without observed interactions as negative items. full means evaluating the model on the set of all items. unixxx, for example uni100, means uniformly sample 100 negative items for each positive item in testing set, and evaluate the model on these positive items with their sampled negative items. popxxx, for example pop100, means sample 100 negative items for each positive item in testing set based on item popularity (Counter(item) in .inter file), and evaluate the model on these positive items with their sampled negative items. Here the xxx must be an integer.
> 
> For explicit feedback (data with label), you should set the mode as labeled and we will evaluate the model based on your label. The default value is full.
> 
> Related issues: [#1103](https://github.com/RUCAIBox/RecBole/issues/1103), [#1178](https://github.com/RUCAIBox/RecBole/issues/1178), [#1256](https://github.com/RUCAIBox/RecBole/issues/1256), [#1260](https://github.com/RUCAIBox/RecBole/issues/1260), [#1288](https://github.com/RUCAIBox/RecBole/discussions/1288), [#1330](https://github.com/RUCAIBox/RecBole/issues/1330), [#1400](https://github.com/RUCAIBox/RecBole/issues/1400), [#1537](https://github.com/RUCAIBox/RecBole/issues/1537)
